### PR TITLE
FPS counter in tutorial game

### PR DESCRIPTION
**Pull request by AdSand**
## Initial setup for the Debug mode.

Uses a debug mode class that contains all the basic functions needed for debug mode.
To use, create a debug mode object in the game, and call the DisplayDebug function, passing the game's dt as a parameter.

Currently only the FPS counter is working in tutorial game

### CODE REVIEW

File: Client/DebugMode.cpp
Comment (DelCherub) [inline Client/DebugMode.cpp]: "Get" suggests that the function will return the FPS count: this should probably be "DisplayFPSCount" instead, likewise for GetMemoryUsage

**Review by DelCherub — COMMENTED** [review]
Looks fine to me apart from changing function names.

File: Client/CMakeLists.txt
Comment (St4rman) [inline Client/CMakeLists.txt]: There's a better way to do this, will send a corresponding message/show in lab. This is a fine way though

File: Client/DebugMode.cpp
Comment (St4rman) [reply Client/DebugMode.cpp]: Agreed

File: Client/TutorialGame.cpp
Comment (St4rman) [inline Client/TutorialGame.cpp:73]: Chad memory managing. Nice one. :D

**Review by St4rman — APPROVED** [review]
Looks relatively okay to me.  I'm sure that there are more stats we'll eventually need to be able to add but this is a decent starting out point. 

Good to merge AFTER merge conflicts solved and changing the method names. 😄

**Review by Odycious — APPROVED** [review]
I have made the requested changes by Matthew and Ed.

**Pull request merged**
